hr_time/public/js/easy_checkin_dialog.js

```diff
@@ -1,129 +1,355 @@
+   * Generates the HTML content for the worklog section of the dialog.
+   * @returns {string} - The HTML content for the worklog section.
+   */
+  getWorklogSectionHTML() {
+    return `<div class="d-flex justify-between">
+                <label id="worklog_section_label">${__("Add Worklog")}</label>
+                <button class="btn btn-outline-info btn-xs edit-full-form-btn">${__(
+                  "Enter complete detail"
+                )}
+                    &nearr;
+                </button>
+            </div>
+            `;
+  }
+
+  /**
+   * Updates the dialog UI based on the selected action.
+   * If "End of work" is selected and no worklogs exist, disables the submit button.
+   */
+  updateDialogBasedOnAction() {
+    let action_value = this.dialogUI.get_value("action");
+    const isEndOfWork = action_value === "End of work";
+    const isIncompleteCheckout = isEndOfWork && !this.hasWorklogs;
+
+    this.dialogUI.get_primary_btn().prop("disabled", isIncompleteCheckout);
+    this.dialogUI.get_field("worklog_section").$wrapper.toggle(isEndOfWork);
+
+    if (this.hasWorklogs) {
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .removeClass("not-filled");
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .addClass("filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .removeClass("not-filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .addClass("filled");
+    } else {
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .addClass("not-filled");
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .removeClass("filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .addClass("not-filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .removeClass("filled");
     }
+  }
 
-    /**
-     * Binds events for numer card of dashboard
-     */
-    static prepare_dashboard() {
-        let dialog = EasyCheckinDialog.singleton()
-
-        document
-            .getElementById("hr_time_number_card_checkin_status")
-            .querySelector(".checkin_status")
-            .onclick = function () {
-            dialog.show();
+  /**
+   * Initializes the dialog with necessary UI adjustments and event bindings.
+   */
+  initializeDialog() {
+    this.dialogUI
+      .get_field("add_log")
+      .$wrapper.addClass("absolute-bottom-right");
+    this.dialogUI.$wrapper
+      .find(".edit-full-form-btn")
+      .click(() => frappe.new_doc("Worklog"));
+    this.dialogUI.$wrapper
+      .find('button[data-fieldname="add_log"]')
+      .addClass(this.hasWorklogs ? "filled" : "not-filled");
+    this.dialogUI.$wrapper
+      .find("label#worklog_section_label")
+      .addClass(this.hasWorklogs ? "filled" : "not-filled");
```

Comment:
The logic for displaying or hiding the text box appears to be defined several times in the class. Please try to outsource the logic in a method.